### PR TITLE
Update MongoDB & Node

### PR DESCRIPTION
**Pull request by StorytellerCZ**
* Update History file
* Update MongoDB to 4.2.7
* Update Node to [12.18.1](https://nodejs.org/en/blog/release/v12.18.1/), which covers the [12.18.0 security release](https://nodejs.org/en/blog/release/v12.18.0/)
* Update npm to 6.14.5

First time updating these dependencies, so not sure if I got everything.

### CODE REVIEW

File: scripts/build-dev-bundle-common.sh [hunk 1]
```diff
@@ -5,10 +5,10 @@ set -u
 
 UNAME=$(uname)
 ARCH=$(uname -m)
-NODE_VERSION=12.16.1
-MONGO_VERSION_64BIT=4.2.5
+NODE_VERSION=12.18.1
+MONGO_VERSION_64BIT=4.2.7
```
Comment (klaussner) [inline scripts/build-dev-bundle-common.sh:9]: We are using a custom build of MongoDB on Linux, so this won't work until we release a build for 4.2.7 here: https://github.com/meteor/mongodb-builder/releases.

Comment (klaussner) [reply scripts/build-dev-bundle-common.sh:9]: MongoDB 4.4 is in the release candidate stage at the moment, so maybe we should wait until 4.4 is released.

Comment (StorytellerCZ) [reply scripts/build-dev-bundle-common.sh:9]: Looks like 4.2.8 has been released recently as well. I think unless it is needless hustle I would go to the most up to date version of MongoDB 4.2.x and focus on 4.4 in the next release, which hopefully will be around the same time.

Comment (klaussner) [reply scripts/build-dev-bundle-common.sh:9]: Ok, that makes sense. However, I currently don't have access to a computer that is fast enough to build MongoDB in a reasonable amount of time. 😄 @filipenevola Does Meteor have a build farm or Linux VM that we can use to build MongoDB?

**Comment by StorytellerCZ** [comment]
@filipenevola Can we at least get a cherrypick for updating Node version to address the June security release: https://nodejs.org/en/blog/vulnerability/june-2020-security-releases/

File: scripts/dev-bundle-tool-package.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ var packageJson = {
   dependencies: {
     // Explicit dependency because we are replacing it with a bundled version
     // and we want to make sure there are no dependencies on a higher version
-    npm: "6.14.0",
+    npm: "6.14.5",
```
Comment (filipenevola) [inline scripts/dev-bundle-tool-package.js:13]:
```suggestion
    npm: "6.14.6",
```

**Comment by filipenevola** [comment]
The MongoDB update I was already going to include on 1.10.3 so let's include these updates as well.

**Comment by filipenevola** [comment]
I'm going to run the tests on 1.10.3 branch.

**Comment by klaussner** [comment]
@filipenevola Did you already build the new MongoDB binaries for Linux?

File: scripts/build-dev-bundle-common.sh (hunk 1 above)
Comment (filipenevola) [reply scripts/build-dev-bundle-common.sh]: hi @klaussner we could add a job to our Jenkins 😉 

I'm trying first locally but it is failing here 

```
error pulling image configuration: Get https://production.cloudflare.docker.com/registry-v2/docker/registry/v2/blobs/sha256/b5/b5b4d78bc90ccd15806443fb881e35b5ddba924e2f475c1071a38a3094c3081d/data?verify=1595591462-MRk%2BlaDsks9fpcIy4%2F6%2By8o2f0A%3D: Service Unavailable
```

do you know if that is a temporary problem with docker registry or if we need to update something here?

**Comment by filipenevola** [comment]
I'm trying, help me here https://github.com/meteor/meteor/pull/11100#discussion_r459989574 😉

File: scripts/build-dev-bundle-common.sh (hunk 1 above)
Comment (klaussner) [reply scripts/build-dev-bundle-common.sh]: Looks like a temporary problem. If I run the container locally, it starts compiling.

**Comment by wreiske** [comment]
I have MongoDB building on a GitHub action son my fork. I'll create a pull request once I have all the pieces put together.

**Comment by wreiske** [comment]
Okay, I got it all working. 

https://github.com/wreiske/mongodb-builder/actions/runs/184631321 should finish in about 2 hours and will automatically attach the finish tgz artifact to the release page here https://github.com/wreiske/mongodb-builder/releases/tag/4.2.8 when it is finished. 

I opted towards cleaning it up and removing the docker specific code, so it may be better to create a separate repository for the GitHub action builder?

**Comment by filipenevola** [comment]
Hi @wreiske I'm ok with your approach, you can submit a PR 👍

**Comment by wreiske** [comment]
> Hi @wreiske I'm ok with your approach, you can submit a PR

@filipenevola https://github.com/meteor/mongodb-builder/pull/1

**Comment by filipenevola** [comment]
@wreiske thank you, we are using your version already 😉

**Pull request merged**